react-components/src/components/RevealToolbar/LayersContainer/Image360LayersContainer.tsx

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import React, { type ReactElement, useState } from 'react';
+import { useReveal } from '../../RevealContainer/RevealContext';
+import { Checkbox, Flex, Menu } from '@cognite/cogs.js';
+import { StyledChipCount, StyledLabel, StyledSubMenu } from './elements';
+import { type Image360Collection } from '@cognite/reveal';
+import uniqueId from 'lodash/uniqueId';
+
+export const Image360CollectionLayerContainer = (): ReactElement => {
+  const viewer = useReveal();
+  const image360Collection = viewer.get360ImageCollections();
+
+  const [selectedImage360Collection, setSelectedImage360Collection] = useState<
+    Array<{ image360: Image360Collection; isToggled: boolean }>
+  >(
+    image360Collection.map((image360Collection) => ({
+      image360: image360Collection,
+      isToggled: true
+    }))
+  );
+
+  const [all360ImagesVisible, setAll360ImagesVisible] = useState(true);
+  const [indeterminate, setIndeterminate] = useState<boolean>(false);
+
+  const count = image360Collection.length.toString();
+
+  const handle360ImagesVisibility = (image360: Image360Collection): void => {
+    selectedImage360Collection.map((data) => {
+      if (data.image360 === image360) {
+        data.isToggled = !data.isToggled;
+        image360.setIconsVisibility(data.isToggled);
+      }
+      return data;
+    });
+    viewer.requestRedraw();
+    setSelectedImage360Collection([...selectedImage360Collection]);
+    setIndeterminate(selectedImage360Collection.some((data) => !data.isToggled));
+    setAll360ImagesVisible(!selectedImage360Collection.every((data) => !data.isToggled));
+  };
+
+  const handleAll360ImagesVisibility = (visible: boolean): void => {
+    selectedImage360Collection.forEach((data) => {
+      data.isToggled = visible;
+      data.image360.setIconsVisibility(data.isToggled);
+    });
```

Comment:
This seems risky, you are mutating the 'state' object directly. I would suggest to either map on the object or create a new copy of the object:

```suggestion
    [...selectedImage360Collection].forEach((data) => {
      data.isToggled = visible;
      data.image360.setIconsVisibility(data.isToggled);
    });
```